nsis/src/nsis/format/NsisBlockHeader.java

```diff
@@ -11,46 +11,34 @@
 
 public class NsisBlockHeader implements StructConverter {
 	private int offset;
-	private int num;
+	private int nbEntries;
+	private final static Structure STRUCTURE;
 
-	public NsisBlockHeader(BinaryReader reader) {
-		setOffset(0);
-		setNum(0);
-
-		try {
-			setOffset(reader.readNextInt());
-			setNum(reader.readNextInt());
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
+	static {
+		STRUCTURE = new StructureDataType("block_header", 0);
+		STRUCTURE.add(DWORD, DWORD.getLength(), "offset", "Offset at which the block header starts");
+		STRUCTURE.add(DWORD, DWORD.getLength(), "num", "Number of entries in the block header");
```

Comment:
num -> entries_number for greater clarity?

Reply:
I used the same name as the spec for reference: https://sourceforge.net/p/nsis/code/HEAD/tree/NSIS/trunk/Source/exehead/fileform.h#l271

See other related comment about nsinst